terraso_backend/apps/soil_id/models/soil_data.py

```diff
@@ -105,6 +106,25 @@ class SlopeSteepness(models.TextChoices):
         blank=True, null=True, validators=[MinValueValidator(0), MaxValueValidator(90)]
     )
 
-    depth_intervals = models.JSONField(
-        blank=True, validators=[validate_depth_intervals], default=default_depth_intervals
+
+class SoilDataDepthInterval(BaseModel, BaseDepthInterval):
+    soil_data = models.ForeignKey(
+        SoilData, on_delete=models.CASCADE, related_name="depth_intervals"
     )
+    label = models.CharField(blank=True, max_length=10)
+
+    class Meta(BaseModel.Meta):
+        constraints = BaseDepthInterval.constraints("soil_data")
+
+    slope_enabled = models.BooleanField(blank=True, default=False)
+    soil_texture_enabled = models.BooleanField(blank=True, default=False)
+    soil_color_enabled = models.BooleanField(blank=True, default=False)
+    vertical_cracking_enabled = models.BooleanField(blank=True, default=False)
+    carbonates_enabled = models.BooleanField(blank=True, default=False)
+    ph_enabled = models.BooleanField(blank=True, default=False)
+    soil_organic_carbon_matter_enabled = models.BooleanField(blank=True, default=False)
+    electrical_conductivity_enabled = models.BooleanField(blank=True, default=False)
+    sodium_adsorption_ratio_enabled = models.BooleanField(blank=True, default=False)
+    soil_structure_enabled = models.BooleanField(blank=True, default=False)
+    land_use_land_cover_enabled = models.BooleanField(blank=True, default=False)
+    soil_limitations_enabled = models.BooleanField(blank=True, default=False)
```

Comment:
suggestion: alphabetize this list.

Reply:
this ordering matches the spec/design. i usually default to chronological ordering (i.e. newest fields at the bottom) for lists like this where this isn't really any sensible grouping/ordering, since it helps with identifying semantic merge conflicts. IMO manually maintaining an alphabetical order isn't always a good idea since a manually maintained alphabetical ordering can easily become an ostensibly alphabetical ordering